src/main/java/io/confluent/idesidecar/restapi/credentials/ApiKeyAndSecret.java

```diff
@@ -106,7 +106,7 @@ public Optional<MultiMap> httpClientHeaders() {
     var headers = MultiMap.caseInsensitiveMultiMap();
     if (key != null || secret != null) {
       // base64 encode the username and password
-      var value = "%s:%s".formatted(key, secret.asCharArray());
+      var value = "%s:%s".formatted(key, secret.asString());
```

Comment:
This bug is super subtle: the toString() of the char array is not the same as a String created from the char array. Added a unit test for this.